Xamarin.Forms.Platform.MacOS/Renderers/CheckBoxRenderer.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using AppKit;
+
+namespace Xamarin.Forms.Platform.MacOS
+{
+	public class CheckBoxRenderer : ViewRenderer<CheckBox, NSButton>
+	{
+		bool _disposed;
+
+		IElementController ElementController => Element;
+
+		protected override void OnElementChanged(ElementChangedEventArgs<CheckBox> e)
+		{
+			if (e.OldElement != null)
+				e.OldElement.CheckedChanged -= OnElementChecked;
+
+			if (e.NewElement != null)
+			{
+				if (Control == null)
+				{
+					SetNativeControl(new NSButton { AllowsMixedState = false, Title = string.Empty });
+
+					Control.SetButtonType(NSButtonType.Switch);
+					Control.Activated += OnControlActivated;
+				}
+
+				UpdateState();
+				e.NewElement.CheckedChanged += OnElementChecked;
+			}
+
+			base.OnElementChanged(e);
+		}
+
+		protected override void Dispose(bool disposing)
+		{
+			if (disposing && !_disposed)
```

Comment:
Same as above - `_disposed` should be set even if `disposing` is false. And if `_disposed` is true you can return early instead of calling `base` again.